fog/block_provider/src/mobilecoind.rs

```diff
@@ -193,10 +194,17 @@ impl BlockProvider for MobilecoindBlockProvider {
         &self,
         tx_out_pub_keys: &[CompressedRistrettoPublic],
     ) -> Result<TxOutInfoByPublicKeyResponse, Error> {
+        let mut meta_builder = MetadataBuilder::new();
+        // We ignore the error here as it only affects distributed tracing and
+        // shouldn't prevent the request from being processed.
+        let _ = meta_builder.inject_context(&Context::current());
+        let call_opt = CallOption::default().headers(meta_builder.build());
         let request = GetTxOutResultsByPubKeyRequest {
             tx_out_public_keys: tx_out_pub_keys.iter().map(|pk| pk.into()).collect(),
         };
-        let response = self.client.get_tx_out_results_by_pub_key(&request)?;
+        let response = self
+            .client
+            .get_tx_out_results_by_pub_key_opt(&request, call_opt)?;
```

Comment:
I realized with the newer block provider trait we might benefit by propagating the context down to the networked block provider.